src/reactor.cc

```diff
@@ -0,0 +1,406 @@
+  ExtractHelium(tritium_storage);
+
+  // Pull excesss tritium out of reserve and put it into storage
+  // Replenish core from Helium Extraction.
+  if (!tritium_reserve.empty() && core_loaded) {
+    double core_deficit = startup_inventory - tritium_core.quantity();
+
+    // Excess tritium will always be in tritium_reserve.
+    double surplus = std::max(
+        tritium_reserve.quantity() - reserve_inventory - core_deficit, 0.0);
+
+    cyclus::Material::Ptr reserve_fuel = tritium_reserve.Pop();
+    cyclus::Material::Ptr core_fuel = tritium_core.Pop();
+
+    core_fuel->Absorb(reserve_fuel->ExtractQty(core_deficit));
+    tritium_storage.Push(reserve_fuel->ExtractQty(surplus));
+
+    RecordOperationalInfo(
+        "Tritium Moved",
+        std::to_string(core_deficit) + "kg of T moved from reserve to core");
+    if (surplus > 0.0) {
+      RecordOperationalInfo(
+          "Tritium Moved",
+          std::to_string(surplus) + "kg of T moved from reserve to storage");
+    }
+    tritium_reserve.Push(reserve_fuel);
+    tritium_core.Push(core_fuel);
+
+    CombineInventory(tritium_storage);
+  }
+
+  // This pulls out some of the blanket each timestep so that fresh blanket can
+  // be added.
+  if (!blanket.empty() &&
+      blanket.quantity() >= blanket_size * blanket_turnover_rate) {
+    cyclus::Material::Ptr blanket_mat = blanket.Pop();
+    cyclus::Material::Ptr spent_blanket =
+        blanket_mat->ExtractQty(blanket_size * blanket_turnover_rate);
+    RecordOperationalInfo(
+        "Blanket Cycled",
+        std::to_string(spent_blanket->quantity()) + "kg of blanket removed");
+    blanket.Push(blanket_mat);
+  } else if (!blanket.empty() &&
+             blanket.quantity() < blanket_size * blanket_turnover_rate) {
+    RecordOperationalInfo(
+        "Blanket Not Cycled",
+        "Total blanket material (" + std::to_string(blanket.quantity()) +
+            ") insufficient to extract " +
+            std::to_string(blanket_size * blanket_turnover_rate) + "kg!");
+  }
+}
+
+void Reactor::Tock() {
+  if (!core_loaded) {
+    try {
+      Startup();
+      fuel_startup_policy.Stop();
+      // blanket_startup_policy.Stop();
+      fuel_refill_policy.Start();
+      core_loaded = true;
+    } catch (const std::exception& e) {
+      RecordOperationalInfo("Startup Error", e.what());
+      LOG(cyclus::LEV_INFO2, "Reactor") << e.what();
+    }
+  }
+
+  CombineInventory(tritium_reserve);
+  CombineInventory(blanket);
+
+  RecordInventories(tritium_core.quantity(), tritium_reserve.quantity(),
+                    tritium_storage.quantity(), blanket.quantity(),
+                    helium_storage.quantity());
+}
+
+void Reactor::EnterNotify() {
+  cyclus::Facility::EnterNotify();
+
+  fuel_startup_policy
+      .Init(this, &tritium_reserve, std::string("Tritium Storage"),
+            &fuel_tracker, std::string("ss"),
+            reserve_inventory + startup_inventory,
+            reserve_inventory + startup_inventory)
```

Comment:
Does this line make the startup requirement equal to reserve+startup inventories? I haven't looked through the docs as much as you, but I thought the startup inventory would count towards the reserve inventory/the reserve inventory is included in the startup inventory

Reply:
I went back and looked at the language that Abdou uses, and it seems like the way to think about it is that the _initial_ storage inventory is the startup, and includes the reserve (that is to say that you are correct). It seems like I misunderstood this when I first started implementation and never corrected it. I think I will need to go back and do a bit of an overhaul on my current three tritium inventories system and squash them into just one to make this more in line with Abdou. I will leave another comment when I've committed that change. This is a good catch, thanks!

Reply:
This has been changed (and holy cow was it a bit bigger of a change than I thought it would be, dang!). I expect this will need another look (not necessarily by you, since you may be gone/busy with other stuff) in the next round of review.